index.md

```diff
@@ -1,16 +1,172 @@
+We should update this class with a more generic name and, in the general case, only support semantic or calendrical versioning.
+That said, we will need to retain the current sorting methods in order to retain backwards compatibility with our current installation.
+We should, however, document that use of anything but semver and calver is not supported for any container other than the Rubin DM stack (and its counterpart for T&S, the SAL stack).
+
+## Compatibility
+
+Of course, for the RSP use case, if we switch to a new version of image construction, we must allow the spawner to continue to spawn older images.
+
+There is a proof-of-concept implementation of such a thing in [the tickets/DM-44731 branch of sciplat-lab][3] which demonstrates this: effectively, we've created symbolic links or copies of various files such that they are present both at the old "everything under `/opt/lsst/software`" location and the newer "Jupyterlab machinery lives in `/usr/local/share/jupyterlab` but the DM pipelines stack is still in `/opt/lsst/software/stack`" locations.  The actual functionality to provide the compatibility layer is found in [the install-compat script run during `docker build`](https://github.com/lsst-sqre/sciplat-lab/blob/tickets/DM-44731/scripts/install-compat).
+
+Minor modifications have been made to the spawner to allow the locations of certain items, like "what command is run inside a user container to start the machinery" to be parameterized in the spawner config, so in either the scenario where all the old-style images are so old we no longer care about supporting them, or the scenario where we're running a completely different payload and therefore have no reason to use the old paths, the spawner can work with only changes to the Phalanx configuration rather than changes to its code.
+
+This also addresses the maintenance issue.
+If we need to rebuild a current version of the container with an antiquated payload, we have the compatibility layer to assist.  The layout of all the non-payload components is free to change, as long as we have a way to link or copy the (fairly few) pieces back into their prior location.
+
+## Proposed Implementation
+
+We think something like [the proof-of-concept implementation][3] is the right direction.  This one is based on the current official Python 3.12 container, which is Debian-based.  It does not produce exactly the same stack container that its counterpart in the lsst-distrib-as-base-container model would, in that the rubin-env Conda environment is freshly installed each time the container runs [lsstinstall][2] and therefore will get newer versions of dependent packages (the decision not to hard-pin versions was taken by the Science Pipelines developers; the author disagrees with their choice, but above my pay grade, I suppose).
```

Comment:
You can choose to hard-pin by using `lsstinstall -X`.

Reply:
Cool!  Thank you.  I'll try that.